azkaban-common/src/main/java/azkaban/executor/ExecutableFlow.java

```diff
@@ -486,26 +487,16 @@ public void setVersionSet(final VersionSet versionSet) {
    * @return Returns the flattened flow props overridden by flow params.
    */
   public void setFlowPropsAndParams(final ProjectLoader projectLoader) {
-    Props props = new Props();
-    // Fetch the flow properties
-    if (FlowLoaderUtils.isAzkabanFlowVersion20(this.azkabanFlowVersion)) {
-      props = FlowLoaderUtils.loadPropsFromYamlFile(projectLoader,
-          this, null);
-    } else {
-      final Map<String, Props> propsMap = projectLoader.
-          fetchProjectProperties(projectId, version);
-      if (null != propsMap) {
-        propsMap.values().forEach(props::putAll);
-      }
-    }
+    Props props = FlowLoaderUtils.isAzkabanFlowVersion20(this.azkabanFlowVersion) ?
+        FlowLoaderUtils.loadPropsFromYamlFile(projectLoader, this, null) :
+        projectLoader.fetchProjectProperty(projectId, version, Constants.PARAM_OVERRIDE_FILE);
 
-    // Filter out the properties to keep only the override ones.
-    Map<String, String> flowOverridePropsMap = props.getMapByPrefix(PARAM_OVERRIDE);
+    // Clone the props object and filter out the properties to keep only the override ones.
+    Map<String, String> flowOverridePropsMap = Props.clone(props).getMapByPrefix(PARAM_OVERRIDE);
```

Comment:
getMapByPrefix is creating map. So we don't need to clone props here. Please double check it. Why to waster memory by cloning.

Reply:
This is to keep a distance from original Props object which was probable cause of memory leak.
Let me know if you still want me to remove it.